### ISSUE TITLE
Style

### ISSUE THREAD

**Issue by DimitriPapadopoulos** [json-parser/json-parser#135]
I have a hard time reading the code.

The code itself is complex, which is OK to obtain a small footprint. It would help understand the code if the style was consistent. Right now, I see for example a mixture of:
```
if (...)
{
   do_something();
```
```
if (...)
{  do_something();
```

These days, I would recommend trying to follow the [Linux kernel coding style](https://www.kernel.org/doc/html/latest/process/coding-style.html). Not that I like it more than others, but it does result in compact and very readable code, and it is widely used. Other widely followed coding styles include the [GTK coding style](https://gitlab.gnome.org/GNOME/gtk/-/blob/master/docs/CODING-STYLE.md) and [GNU coding standards](https://www.gnu.org/prep/standards/html_node/Writing-C.html).

Current style is OK too (3 space for indents for example) but it needs to be consistent and consistently applied.

**Comment by LB--** [comment]
I agree that there should be some more consistency, though I'd prefer to enact more consistency in the least invasive way so that the line attribution is still useful in most cases. Whitespace-only changes are fine because git has an option to ignore them when obtaining line attribution, but I'm not sure how it handles e.g. moving from the second snippet to the first snippet in your example, I'll have to check that.
